src/ensembl/src/content/app/species-selector/state/speciesSelectorActions.ts

```diff
@@ -143,175 +140,163 @@ export const clearSelectedSearchResult = createAction(
 //   }
 // };
 
-export const ensureSpeciesIsCommitted = (
-  genomeId: string
-): ThunkAction<void, any, null, Action<string>> => async (
-  dispatch,
-  getState: () => RootState
-) => {
-  const state = getState();
-  const committedSpecies = getCommittedSpecies(state);
-  const genomeInfo = getGenomeInfoById(state, genomeId);
-  if (getCommittedSpeciesById(state, genomeId) || !genomeInfo) {
-    return;
-  }
-
-  const newCommittedSpecies = [
-    ...committedSpecies,
-    {
-      ...pickBy(genomeInfo, (value, key) => {
-        return key !== 'example_objects';
-      }),
-      isEnabled: true
+export const ensureSpeciesIsCommitted =
```

Comment:
How about changing this too: `ensureSpeciesIsCommitted` -> `areSpeciesCommitted`?

Reply:
This action is not a question 😃

Reply:
@azangru I didn't check the entire function properly and where it is used. My bad. However, why have we prefixed the function with `ensure` rather than `save`?

Reply:
Hard to explain 😄  My intention in the naming of this function was: in certain situations (for example when user lands on a page by entering a url in the address bar) a species can be already committed, but we cannot be sure, but we really need it to be. Same with a neighbouring function: a species may be already enabled, but we cannot be sure, but we want to make sure that it is.